deployment/deploy_to_preview.sh

```diff
@@ -8,9 +8,11 @@ chmod 600 ~/.ssh/id_rsa
 timeout 300 bash -c "while ! ssh -o ConnectTimeout=3 ubuntu@3.71.162.94 exit; do echo 'target server not yet there - retrying in 1s'; sleep 1; done" || exit
 
 location=/home/ubuntu/dataland
-#shut down currently running dataland application and purge files on server
+# shut down currently running dataland application and purge files on server
 ssh ubuntu@3.71.162.94 "cd $location && sudo docker-compose down; rm -rf $location; mkdir -p $location/jar"
+# make sure no remnants remain when docker-compose file changes
+ssh ubuntu@3.71.162.94 'sudo docker kill $(sudo docker ps -q); sudo docker rm $(sudo docker ps -a -q); sudo docker system prune --force; sudo docker info'
 
-scp -r ./dataland-frontend/dist ./deployment/docker-compose.yml ./nginx.conf ubuntu@3.71.162.94:$location
+scp -r ./dataland-frontend/dist ./deployment/docker-compose.yml ./dataland-inbound-proxy/ ubuntu@3.71.162.94:$location
 scp ./dataland-backend/build/libs/dataland-backend*.jar ubuntu@3.71.162.94:$location/jar/dataland-backend.jar
-ssh ubuntu@3.71.162.94 "export SKYMINDER_URL=$SKYMINDER_URL; export SKYMINDER_PW=$SKYMINDER_PW; export SKYMINDER_USER=$SKYMINDER_USER; cd $location; sudo -E docker-compose up -d"
+ssh ubuntu@3.71.162.94 "cd $location; docker-compose pull; SKYMINDER_URL=$SKYMINDER_URL SKYMINDER_PW=$SKYMINDER_PW SKYMINDER_USER=$SKYMINDER_USER sudo -E docker-compose up -d"
```

Comment:
Was genau ist der Gewinn zuerst den `docker-compse pull` zu machen?
Laut Internet garantiert das, dass die neuesten Versionen der verwendeten Images gezogen werden. 
In Zeile 14 werden aber doch alle container removet. Und mit `docker system prun` auch alle Netzwerke und Images.
Dadurch muss doch `docker-compose up -d` sowieso alles neu ziehen.

Reply:
Ok `docker system prune` entfernt nur dangling und unreferenced images.  D.h. die durch die Services im docker compose file benamten Images werden nicht entfernt.
Somit macht es doch Sinn, die Images neu zu ziehen. Richtig?

Reply:
Alles klar, Danke 👍